### PR TITLE
RavenDB-12823 should mutate the revision flags to include `Conflicted…

### PR DESCRIPTION
**Pull request by karmeli87**
…` once conflict is resolved

### CODE REVIEW

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 1]
```diff
@@ -721,10 +723,13 @@ private static void DeleteCountOfRevisions(DocumentsOperationContext context, Sl
 
             using (Slice.From(context.Allocator, changeVector, out var changeVectorSlice))
             {
-                var revisionExists = table.VerifyKeyExists(changeVectorSlice);
+                var revisionExists = table.ReadByKey(changeVectorSlice,out var tvr);
                 // Revisions are immutable.
```
Comment (ayende) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs]: wrong comment?

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 2]
```diff
@@ -769,6 +774,52 @@ private static void DeleteCountOfRevisions(DocumentsOperationContext context, Sl
             }
         }
 
+        private void MarkRevisionsAsConflictedIfNeeded(DocumentsOperationContext context, Slice lowerId, Slice idSlice, DocumentFlags flags, TableValueReader tvr, Table table,
+            Slice changeVectorSlice)
+        {
+            // Revisions are immutable, but if there was a conflict we need to update the flags accordingly with the `Conflicted` flag.
```
Comment (ayende) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:779]: Why is that the case? The revision itself isn't conflicted, after all. 
Or is it that it was conflicted on another node?

Comment (karmeli87) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:779]: This is relevant in the case when revisions are enabled.
So node A generates a document with a revision, and node B also.
When the conflict is resolved, we need to add the `Conflicted` flag to an already existing revision.

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs]: yes, removed

**Pull request merged**